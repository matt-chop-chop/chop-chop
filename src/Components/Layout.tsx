import { Container, Flex } from "@chakra-ui/react";
import { Header, Footer } from "@/Components";

type LayoutProps = {
  children: React.ReactNode;
};

const Layout = ({ children }: LayoutProps) => {
  return (
    <Flex
      flexDirection="column"
      height="100vh"
      justifyContent="space-between"
      width="100%"
    >
      <Flex flexDirection="column" width="100%">
        <Container centerContent maxWidth="100%" p={0}>
          <Header />
        </Container>

        <Container
          centerContent
          maxWidth="100%"
          py={["108px", "108px", "140px"]}
        >
          <Flex flexDirection="column" width="100%">
            {children}
          </Flex>
        </Container>
      </Flex>

      <Container m={0} maxWidth="100%" p={0}>
        <Footer />
      </Container>
    </Flex>
  );
};

export default Layout;
