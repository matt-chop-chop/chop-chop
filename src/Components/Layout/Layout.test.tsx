import { shallow } from "enzyme";
import React from "react";
import Layout from "./Layout";

describe("Layout", () => {
  it("is exported from Layout", () => {
    expect(Layout).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(
      <Layout>
        <div>test</div>
      </Layout>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
