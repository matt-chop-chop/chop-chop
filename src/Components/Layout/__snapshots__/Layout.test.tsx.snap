// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders as expected  1`] = `
<Flex
  flexDirection="column"
  height="100vh"
  justifyContent="space-between"
  width="100%"
>
  <Flex
    flexDirection="column"
    width="100%"
  >
    <Container
      centerContent={true}
      maxWidth="100%"
      p={0}
      width="100%"
    >
      <Header />
    </Container>
    <Container
      centerContent={true}
      maxWidth="1320px"
      py={
        [
          "108px",
          "108px",
          "140px",
        ]
      }
    >
      <Flex
        flexDirection="column"
        width="100%"
      >
        <div>
          test
        </div>
      </Flex>
    </Container>
  </Flex>
  <Container
    centerContent={true}
    maxWidth="1320px"
    p={0}
    width="100%"
  >
    <Footer />
  </Container>
</Flex>
`;
